### PR TITLE
No issue: Add TabPartition tests

### PR DESCRIPTION
**Pull request by rocketsroger**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/browser/state/src/test/java/mozilla/components/browser/state/state/TabPartitionTest.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+class TabPartitionTest {
+
+    @Test
+    fun `GIVEN a null tab partition THEN tab partition is empty`() {
+        val tabPartition: TabPartition? = null
+
+        assertEquals(tabPartition.isEmpty(), true)
```
Comment (csadilek) [inline components/browser/state/src/test/java/mozilla/components/browser/state/state/TabPartitionTest.kt:16]: nit: `isEmpty` and `isNotEmpty` return `Boolean` so should be able to use `assertTrue`/`assertFalse` for all of these.

File: components/browser/state/src/test/java/mozilla/components/browser/state/state/TabPartitionTest.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+        assertEquals(tabPartition.isEmpty(), false)
+        assertEquals(tabPartition.isNotEmpty(), true)
+    }
+
+    @Test
+    fun `GIVEN a tab partition with non-empty tab group THEN get group by name will return the group`() {
+        val tabPartition = TabPartition("test", listOf(TabGroup("test id", "abc", listOf("tab1", "tab2"))))
+
+        assert(tabPartition.getGroupByName("abc") != null)
```
Comment (csadilek) [inline components/browser/state/src/test/java/mozilla/components/browser/state/state/TabPartitionTest.kt:48]: That's a JDK `assert` not a junit one. `assertTrue`.

File: components/browser/state/src/test/java/mozilla/components/browser/state/state/TabPartitionTest.kt [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        assertEquals(listOf("tab1", "tab2"), tabPartition.getGroupByName("abc")?.tabIds)
+        assertEquals(tabPartition.isEmpty(), false)
+        assertEquals(tabPartition.isNotEmpty(), true)
+    }
+
+    @Test
+    fun `GIVEN a tab partition with non-empty tab group THEN get group by ID will return the group`() {
+        val tabPartition = TabPartition("test", listOf(TabGroup("test id", "abc", listOf("tab1", "tab2"))))
+
+        assert(tabPartition.getGroupById("test id") != null)
```
Comment (csadilek) [inline components/browser/state/src/test/java/mozilla/components/browser/state/state/TabPartitionTest.kt:58]: That's a JDK `assert` not a junit one. `assertTrue`.

Comment (rocketsroger) [reply components/browser/state/src/test/java/mozilla/components/browser/state/state/TabPartitionTest.kt:58]: Thanks for catching these.  I'll update.

**Review by csadilek — APPROVED** [review]

**Pull request merged**